### PR TITLE
fix(embedded): use tmp folder for unit test cases

### PR DESCRIPTION
**Pull request by arriqaaq**

### CODE REVIEW

File: embedded/appendable/multiapp/multi_app_test.go [hunk 1]
```diff
@@ -165,8 +171,11 @@ func TestMultiAppClosedFiles(t *testing.T) {
 }
 
 func TestMultiAppReOpening(t *testing.T) {
-	a, err := Open("testdata", DefaultOptions().WithFileSize(1))
-	defer os.RemoveAll("testdata")
+	path, err := ioutil.TempDir(os.TempDir(), "testdata")
+	require.NoError(t, err)
+	defer os.RemoveAll(path)
+
+	a, err := Open(path, DefaultOptions().WithFileSize(1))
```
Comment (jeroiraz) [inline embedded/appendable/multiapp/multi_app_test.go:178]: `testdata_copy` below should also be a temp folder

File: embedded/appendable/remoteapp/remote_app_test.go [hunk 2]
```diff
@@ -835,8 +854,7 @@ func TestRemoteStorageDownloadUnrecoverableError(t *testing.T) {
 }
 
 func TestRemoteStorageOpenChunkWhenUploading(t *testing.T) {
-	defer os.RemoveAll("testdata")
-	prepareLocalTestFiles(t)
+	path := prepareLocalTestFiles(t)
```
Comment (jeroiraz) [inline embedded/appendable/remoteapp/remote_app_test.go:857]:
```suggestion
	path := prepareLocalTestFiles(t)
	defer os.RemoveAll(path)
```

File: embedded/store/immustore_test.go [hunk 3]
```diff
@@ -306,13 +306,13 @@ func TestImmudbStoreEdgeCases(t *testing.T) {
 		require.EqualError(t, err, "stat invalid\x00_dir_name: invalid argument")
 	})
 
-	t.Run("should fail with permiission denied", func(t *testing.T) {
```
Comment (jeroiraz) [inline embedded/store/immustore_test.go:309]: is this test case failing in the github runner? Recent changes shouldn't affect this edge case... However I was able to reproduce it locally when running test inside a container but with macOS as host

File: pkg/server/server_test.go [hunk 4]
```diff
@@ -593,6 +593,7 @@ func TestServerLoaduserDatabase(t *testing.T) {
 }
 
 func TestServerLoadUserDatabases(t *testing.T) {
+	os.RemoveAll("./data_v1.1.0")
```
Comment (jeroiraz) [inline pkg/server/server_test.go:596]: a tmp folder could be used here as well

File: embedded/store/immustore_test.go (hunk 3 above)
Comment (arriqaaq) [reply embedded/store/immustore_test.go:309]: this test is only failing on a mac, and also in ubuntu container. I will uncomment this and run in gitlab runner

File: embedded/appendable/remoteapp/remote_app_test.go (hunk 2 above)
Comment (arriqaaq) [reply embedded/appendable/remoteapp/remote_app_test.go:857]: ack

File: embedded/appendable/multiapp/multi_app_test.go (hunk 1 above)
Comment (arriqaaq) [reply embedded/appendable/multiapp/multi_app_test.go:178]: ack

File: pkg/server/server_test.go (hunk 4 above)
Comment (arriqaaq) [reply pkg/server/server_test.go:596]: for some reason it doesn't work as expected with tmp folder
